[KeypressListener] Adds support for `useCapture` and `options` (#5250)
Shopify/polaris: UNRELEASED.md

### Enhancements

> - Added `useCapture` and `options` props in `KeypressListener` to allow passing through those options to the underlying `addEventListener` call ([#5221](https://github.com/Shopify/polaris-react/pull/5221))

Reviewer: ```suggestion
- Added `useCapture` and `options` props to `KeypressListener` to support passing options to the underlying `addEventListener` call ([#5221](https://github.com/Shopify/polaris-react/pull/5221))
```
